cache/fast_lru_cache.cc

```diff
@@ -9,26 +9,38 @@
 
 #include "cache/fast_lru_cache.h"
 
+#include <math.h>
+
 #include <cassert>
 #include <cstdint>
 #include <cstdio>
+#include <functional>
 
 #include "monitoring/perf_context_imp.h"
 #include "monitoring/statistics.h"
 #include "port/lang.h"
+#include "util/hash.h"
 #include "util/mutexlock.h"
 
-#define KEY_LENGTH \
-  16  // TODO(guido) Make use of this symbol in other parts of the source code
-      // (e.g., cache_key.h, cache_test.cc, etc.)
+#define LOAD_FACTOR 0.7
 
 namespace ROCKSDB_NAMESPACE {
 
 namespace fast_lru_cache {
 
+namespace {
+// Returns x % 2^{bits}.
+inline uint32_t BinaryMod(uint32_t x, uint32_t bits) {
+  return (x << (32 - bits)) >> (32 - bits);
+}
+}  // namespace
+
 LRUHandleTable::LRUHandleTable(int hash_bits)
     : length_bits_(hash_bits),
-      list_(new LRUHandle* [size_t{1} << length_bits_] {}) {}
+      occupancy_(0),
+      array_(new LRUHandle[size_t{1} << length_bits_]) {
```

Comment:
It looks like a proper initializing constructor for LRUHandle would save a lot of awkward/fragile code in places. Like here the explicit `memset` can be avoided. (Since C++11, if `Blah` is a class or struct, `(new Blah[n])` always applies the Blah default constructor to all n elements.)